SHA256_Init is deprecated

```
skymp/skymp5-server/cpp/server_guest_lib/database_drivers/MongoDatabase.cpp:304:3: warning: 'SHA256_Init' is deprecated [-Wdeprecated-declarations]
  SHA256_Init(&sha256);
  ^
skymp/build/vcpkg_installed/x64-linux/include/openssl/sha.h:73:1: note: 'SHA256_Init' has been explicitly marked deprecated here
OSSL_DEPRECATEDIN_3_0 int SHA256_Init(SHA256_CTX *c);
^
```

--- a/skymp5-server/cpp/server_guest_lib/database_drivers/MongoDatabase.cpp
+++ b/skymp5-server/cpp/server_guest_lib/database_drivers/MongoDatabase.cpp
@@ -300,11 +300,7 @@ std::string MongoDatabase::BytesToHexString(const uint8_t* bytes,
 std::string MongoDatabase::Sha256(const std::string& str)
 {
   uint8_t hash[SHA256_DIGEST_LENGTH];
-  SHA256_CTX sha256;
-  SHA256_Init(&sha256);
-  SHA256_Update(&sha256, str.c_str(), str.size());
-  SHA256_Final(hash, &sha256);
-
+  SHA256(reinterpret_cast<const uint8_t*>(str.data()), str.size(), hash);
   return BytesToHexString(hash, SHA256_DIGEST_LENGTH);
 }
 

